eternalcore-core/src/main/java/com/eternalcode/core/placeholder/AsyncPlaceholderValue.java

```diff
@@ -0,0 +1,29 @@
+package com.eternalcode.core.placeholder;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+public class AsyncPlaceholderValue<T> {
+    private final CompletableFuture<T> future;
+    private final Function<T, String> formatter;
+
+    public AsyncPlaceholderValue(CompletableFuture<T> future, Function<T, String> formatter) {
+        this.future = future;
+        this.formatter = formatter;
+    }
+
+    @Override
+    public String toString() {
```

Comment:
I don't think that this name is good. I would suggest change to maybe "getValue" or smth like this.
```suggestion
    public String getValue() {
```